src/util/string_constant.cpp

```diff
@@ -87,13 +87,12 @@ bool string_constantt::from_array_expr(const array_exprt &src)
 
   std::string value;
 
-  forall_operands(it, src)
+  for(const auto &op : src.operands())
   {
-    mp_integer int_value=0;
-    if(to_integer(*it, int_value))
+    if(!op.is_constant())
       return true;
-    unsigned unsigned_value=integer2unsigned(int_value);
-    value+=static_cast<char>(unsigned_value);
+
+    value += numeric_cast_v<char>(op);
   }
```

Comment:
I am a bit worried about this one -- the previous implementation would have done the conversion, say from -1 to 255; now you get an invariant failure.

Reply:
Do negative numbers make sense in this context? I'm happy to amend this in any reasonable way. The code does not seem to be used anywhere, so I wouldn't know how to test, I'm afraid.

Reply:
#3168 takes care of this!

Reply:
Ok, I will rebase once that is merged.